lib/opam.ml

```diff
@@ -238,3 +238,29 @@ module Pos = struct
     let pos = from_value value in
     errorf ~pos fmt
 end
+
+module Value = struct
+  module String = struct
+    let from_value value =
+      match (value : OpamParserTypes.FullPos.value) with
```

Comment:
Maybe this would be better to attach to the line above? Or as part of the qualification of the record label?

Reply:
I guess this is mostly a matter of personal preferences I tend to do this to avoid having to disambiguate constructors or record fields in the first pattern (which I personally find kind of ugly looking) or having to use a local open. This makes it clear that we simply want to pattern match over the concrete type.

We could move it to the function argument if you prefer, I don't particularly mind. I just find it quite nice to do it here as it is possible to consistently annotate in the pattern match rather than doing it elsewhere since we're not always pattern matching over one of the argument.